### PR TITLE
utils: add test for SelectPeers

### PR DESCRIPTION
**Pull request by hasyimibhar**
Added `TestSelectPeers`. Also added test utility `newNode`, which might be useful for other tests.

### CODE REVIEW

File: utils_test.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+		}
+	}()
+
+	return client, addr, func() {
+		server.GracefulStop()
+		cleanup()
+	}
+}
+
+func newKV(kv string, path string) (store.KV, func()) {
```
Comment (iwasaki-kenta) [inline utils_test.go]: I think we should expose this as a function within the `store` pkg. Thoughts?

Comment (hasyimibhar) [reply utils_test.go]: Yeah we should. I actually copy-pasted this from the `avl` package: https://github.com/perlin-network/wavelet/blob/master/avl/tree_test.go#L431-L454

Comment (iwasaki-kenta) [reply utils_test.go]: Awesome, care to do that in this PR as well?

Comment (hasyimibhar) [reply utils_test.go]: Alright sure.

**Pull request merged**